app/controllers/api/v1/authorization_definitions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class API::V1::AuthorizationDefinitionsController < API::V1Controller
+  before_action -> { doorkeeper_authorize! :read_authorizations }, only: %i[index show]
+  before_action :set_authorization_definition, only: %i[show]
+
+  def index
+    authorization_definitions = AuthorizationDefinition.all.select do |definition|
+      current_user_authorization_request_types.include?(definition.authorization_request_class.to_s)
+    end
+
+    render json: authorization_definitions,
+      each_serializer: API::V1::AuthorizationDefinitionSerializer,
+      status: :ok
+  end
+
+  def show
+    render json: @authorization_definition,
+      serializer: API::V1::AuthorizationDefinitionSerializer,
+      status: :ok
+  end
+
+  private
+
+  def set_authorization_definition
+    @authorization_definition = AuthorizationDefinition.all.find do |definition|
+      definition.id == params[:id] &&
+        current_user_authorization_request_types.include?(definition.authorization_request_class.to_s)
```

Comment:
```suggestion
    @authorization_definition = AuthorizationDefinition.where(authorization_request_class: current_user_authorization_request_types).find(params[:id])
```

même remarque qu'au dessus. limite `find_by` serait aussi mieux mais bon.. 
le `find` devrait lever un `ActiveRecord::NotFoundError` aussi

Reply:
can do